include/nnvm/top/tensor.h

```diff
@@ -84,6 +89,14 @@ struct CastParam : public dmlc::Parameter<CastParam> {
   }
 };
 
+struct IndicatorParam : public dmlc::Parameter<IndicatorParam> {
```

Comment:
can we simply use greater and less as numpy did?

Reply:
I think keeping indicator function is more useful here:
1. Unified interface in core src here is more concise as they share lots of properties including same gradient function. 
2. The operator can be potentially extended to generate mask array based on rules like equal to some value or max over certain axises. I also took a look at less and greater operator, it can be composed of broadcast sub and indicator function.

Reply:
Thanks for digging deep this issue. Can you give an example of what do you mean by concise?

In general, we should avoid creating functions with user-facing names (like name indicator) when its application is not user-facing (depend on flags that user don't like to wrirte). For non-user facing functions, it is always a good idea to use underscore. This is a good way to avoid legacy issues later.

On the other hand, we should avoid creating underscore functions at all when possible and reuse existing user-facing functions, as likely high order gradient will be well defined.

These are general guidelines and I would like to discuss specific details in here

Reply:
I agree with what you say, passing flags to operator is not user friendly. 
By concise, I mean "implementation-friendly" :)
I will change the code to align with numpy. Thank you for review.

Reply:
For greater, less. The axis and exclude parameter are not consistent with numpy. Should they be removed, or rename the function with an underscore.

Reply:
For now, it's not used. I changed them to numpy compatible.